src/hotspot/share/services/mallocHeader.hpp

```diff
@@ -87,6 +87,13 @@ class outputStream;
  *   the bytes are stored individually.
  */
 
+// Contains all of the necessary data to record a free().
+struct FreePackage {
+  const size_t size;
+  const MEMFLAGS flags;
+  const uint32_t mst_marker;
+};
```

Comment:
Could this be nested into MallocHeader, and renamed to something like "FreeInfo"?